pkg/controller/migration/namespace_migration.go

```diff
@@ -384,19 +447,24 @@ func (m *CoreNamespaceMigration) ensureKubeSysNodeDaemonSetHasNodeSelectorAndIsR
 		if err != nil {
 			return false, err
 		}
-		if ds.Status.DesiredNumberScheduled != ds.Status.NumberReady || ds.Status.ObservedGeneration != ds.ObjectMeta.Generation {
-			return false, fmt.Errorf("not all pods are ready yet: %d/%d", ds.Status.NumberReady, ds.Status.DesiredNumberScheduled)
+		if ds.Status.ObservedGeneration != ds.ObjectMeta.Generation {
+			log.Info(fmt.Sprintf("waiting for observed generation (%d) to match object generation (%d)", ds.Status.ObservedGeneration, ds.ObjectMeta.Generation))
+			return false, nil
 		}
+		if ds.Status.DesiredNumberScheduled != ds.Status.NumberReady {
+			log.Info(fmt.Sprintf("waiting for kube-system/calico-node to have %d replicas, currently at %d", ds.Status.DesiredNumberScheduled, ds.Status.NumberReady))
+			return false, nil
```

Comment:
We used to return an error here, which would go all the way back to the Reconcile() retry. I switched it to return false/nil to simply retry in the PollImmediate - is there a reason this is no good? It means less work overall, and fewer scary looking log messages during an "expected" transition.

Reply:
I want to say it was so that if any new nodes showed up that they would get labeled but I'm not seeing why that would be necessary here.